source/agora/test/Byzantine.d

```diff
@@ -0,0 +1,208 @@
+/*******************************************************************************
+
+    Contains Byzantine node tests, which refuse to co-operate in the
+    SCP consensus protocol in various ways.
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.test.Byzantine;
+
+version (unittest):
+
+import agora.api.Validator;
+import agora.common.Config;
+import agora.common.Task;
+import agora.common.Types;
+import agora.common.crypto.Key;
+import agora.common.crypto.Schnorr;
+import agora.consensus.data.Block;
+import agora.consensus.data.ConsensusParams;
+import agora.consensus.data.Transaction;
+import agora.consensus.protocol.Nominator;
+import agora.network.Clock;
+import agora.network.NetworkClient;
+import agora.network.NetworkManager;
+import agora.node.Ledger;
+import agora.test.Base;
+
+import scpd.types.Stellar_SCP;
+
+import geod24.Registry;
+
+import std.algorithm;
+import std.exception;
+import std.format;
+import std.range;
+import std.stdio;
+
+import core.exception;
+import core.stdc.time;
+
+enum ByzantineReason 
+{
+    none,
+    not_signing_envelope,
+    bad_signing_envelope
+}
+
+extern(C++) class ByzantineNominator : TestNominator
+{
+    private ByzantineReason reason;
+    
+    extern(D) this (immutable(ConsensusParams) params,
+        Clock clock, NetworkManager network, KeyPair key_pair, Ledger ledger,
+        TaskManager taskman, ByzantineReason reason)
+    {
+        this.reason = reason;
+        super(params, clock, network, key_pair, ledger, taskman,
+            this.txs_to_nominate);
+    }
+
+    // refuse to sign
+    extern(C++) override void signEnvelope (ref SCPEnvelope envelope)
+    {
+        switch (reason) 
+        {
+            bad_signing_envelope:
```

Comment:
You are missing `case:` here. You've created a label instead of a `case` statement.

Reply:
Updated.